app/src/main/java/com/cornellappdev/volume/ui/components/general/IndividualPublicationComponents.kt

```diff
@@ -1,4 +1,287 @@
+            maxLines = 6,
+            overflow = TextOverflow.Ellipsis,
+            fontFamily = notoserif,
+            fontWeight = FontWeight.Medium,
+            fontSize = 14.sp
+        )
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(top = 10.dp, start=12.dp, end=12.dp), horizontalArrangement = Arrangement.Center
+        ) {
+            for (social in publication.socials) {
+                if (social.social == "instagram") {
+                    Image(
+                        modifier = Modifier
+                            .scale(1.3f),
+                        painter = painterResource(R.drawable.ic_instagram),
+                        contentDescription = null,
+                    )
+                    HyperlinkText(
+                        fullText = "Instagram",
+                        modifier = Modifier.padding(start = 10.dp),
+                        hyperLinks = mutableMapOf(
+                            "Instagram" to social.url
+                        ),
+                        textStyle = TextStyle(
+                            textAlign = TextAlign.Center,
+                            color = VolumeOrange
+                        ),
+                    )
+                }
+                if (social.social == "facebook") {
+                    Image(
+                        modifier = Modifier
+                            .scale(1.3f),
+                        painter = painterResource(R.drawable.ic_facebook),
+                        contentDescription = null,
+                    )
+                    HyperlinkText(
+                        fullText = "Facebook",
+                        modifier = Modifier.padding(start = 10.dp),
+                        hyperLinks = mutableMapOf(
+                            "Facebook" to social.url
+                        ),
+                        textStyle = TextStyle(
+                            textAlign = TextAlign.Center,
+                            color = VolumeOrange
+                        ),
+                    )
+                }
+            }
+            Image(
+                modifier = Modifier
+                    .padding(start = 10.dp)
+                    .scale(1.3f),
+                painter = painterResource(R.drawable.ic_link),
+                contentDescription = null,
+            )
+            HyperlinkText(
+                fullText = publication.name,
+                modifier = Modifier.padding(start = 10.dp),
+                hyperLinks = mutableMapOf(
+                    publication.name to publication.websiteURL
+                ),
+                textStyle = TextStyle(
+                    textAlign = TextAlign.Center,
+                    color = VolumeOrange
+                ),
+            )
+        }
+
+    }
+}
+
+@Composable
+fun HyperlinkText(
+    modifier: Modifier = Modifier,
+    fullText: String,
+    hyperLinks: Map<String, String>,
```

Comment:
I see as well that you only call this function with one entry to the map, this can be simplified by using a simpler data structure: [Pair](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-pair/)